charms/istio-pilot/src/charm.py

```diff
@@ -181,6 +201,56 @@ def _get_image_config(self):
         image_config = yaml.safe_load(self.model.config[IMAGE_CONFIGURATION])
         return image_config
 
+    # ---- Start of the block
+    # ---- WARNING: this feature is not recommended, but is supported in 1.17 and 1.18.
+    # ---- For details please refer to canonical/istio-operators#380.
+    # ---- FIXME: Remove this block after releasing 1.18.
+    def remove_tls_secret(self, event) -> None:
+        """Remove the secret that saves TLS information and reconcile the ingress gateway."""
+        try:
+            secret = self.model.get_secret(label="istio-tls-secret")
+            secret.remove_all_revisions()
+            self._reconcile_gateway()
```

Comment:
Do we need to `._reconcile_gateway()` here, or can we just rely on the [observed secret events](https://github.com/canonical/istio-operators/pull/394/files#diff-d3a4e684bd1a3a1822df5ab7605cfce00c97cd7b52f44ed770526c4f4f80eacdR128) to trigger the correct reconciliation?

Reply:
We need it because there is no `secret created` event, so initially when first creating the secret, nothing will be observed.